browser/ui/views/tabs/features.cc

```diff
@@ -79,8 +83,22 @@ std::pair<int, int> GetLeadingTrailingCaptionButtonWidth(
   // On Mac, window caption buttons are drawn by the system.
   return {80, 0};
 #elif BUILDFLAG(IS_LINUX)
-  // On Linux, we can't overlay caption buttons on toolbar.
-  return {0, 0};
+  auto* browser_view =
+      BrowserView::GetBrowserViewForNativeWindow(frame->GetNativeWindow());
+  DCHECK(browser_view);
+
+  auto* profile = browser_view->browser()->profile();
+  auto* linux_ui_theme = ui::LinuxUiTheme::GetForProfile(profile);
+  auto* theme_service_factory = ThemeServiceFactory::GetForProfile(profile);
+  if (linux_ui_theme && theme_service_factory->UsingSystemTheme()) {
+    // Uses gtk-provided caption buttons. We size them as big as toolbar buttons
+    return {0, (GetLayoutConstant(TOOLBAR_BUTTON_HEIGHT) +
+                GetLayoutInsets(TOOLBAR_BUTTON).width()) *
```

Comment:
The margin between the buttons looks kind of big to me:
![image](https://user-images.githubusercontent.com/7678024/218848803-5f914548-e2f4-45f6-8279-369aa0ddc6f9.png)

Reply:
This is hovering over minimize:
![image](https://user-images.githubusercontent.com/7678024/218849682-17de5a8b-aee2-4b2a-9a7f-c9b5b9db9045.png)

Reply:
I almost feel like you don't need the spacing between buttons because they have so much padding :laughing: